src/sema/eval.rs

```diff
@@ -196,3 +201,186 @@ pub fn eval_const_rational(
         )),
     }
 }
+
+fn eval_constants_in_expression(
+    expr: &Expression,
+    ns: &mut Namespace,
+    results: &mut Vec<BigInt>,
+) -> Expression {
+    match expr {
+        Expression::Add(loc, ty, unchecked, left, right) => {
+            let left = eval_constants_in_expression(left, ns, results);
+            let right = eval_constants_in_expression(right, ns, results);
+
+            if let (Expression::NumberLiteral(_, _, left), Expression::NumberLiteral(_, _, right)) =
+                (&left, &right)
+            {
+                results.push(left.add(right));
+                Expression::NumberLiteral(*loc, ty.clone(), left.add(right))
+            } else {
+                Expression::Add(
+                    *loc,
+                    ty.clone(),
+                    *unchecked,
+                    Box::new(left),
+                    Box::new(right),
+                )
+            }
+        }
+        Expression::Subtract(loc, ty, unchecked, left, right) => {
+            let left = eval_constants_in_expression(left, ns, results);
+            let right = eval_constants_in_expression(right, ns, results);
+
+            if let (Expression::NumberLiteral(_, _, left), Expression::NumberLiteral(_, _, right)) =
+                (&left, &right)
+            {
+                results.push(left.sub(right));
+                Expression::NumberLiteral(*loc, ty.clone(), left.sub(right))
+            } else {
+                Expression::Subtract(
+                    *loc,
+                    ty.clone(),
+                    *unchecked,
+                    Box::new(left),
+                    Box::new(right),
+                )
+            }
+        }
+
+        Expression::Multiply(loc, ty, unchecked, left, right) => {
+            let left = eval_constants_in_expression(left, ns, results);
+            let right = eval_constants_in_expression(right, ns, results);
+
+            if let (Expression::NumberLiteral(_, _, left), Expression::NumberLiteral(_, _, right)) =
+                (&left, &right)
+            {
+                results.push(left.mul(right.to_u32().unwrap()));
```

Comment:
How do you know the number fits in a `u32`? Numbers can be negative too.
Keep in mind that arithmetic operations in Solidity support up to `u256`.